common/src/stack/checklist/pylib/__init__.py

```diff
@@ -59,13 +59,33 @@ def isPostPkgInstallStage(self):
 	# through the State st.
 	#
 	def isKnownState(self, st):
-		arr = self.stateArr
-		for a in arr:
+		for a in self.stateArr:
 			if a.state == st and not a.isError:
 				return True
 
 		return False
 
+	#
+	# Find index of State  in Backend StateMessage array
+	# and return max int value if it does not exist
+	#
+	def findStateMsgIndex(self, st):
+		for idx, a in enumerate(self.stateArr):
+			if a.state == st:
+				return idx
+		return 1000
+
+	#
+	# Check if this message has been seen before and returns
+	# True if this is the case.
+	#
+	def hasStateMessage(self, sm):
+		for a in self.stateArr:
+			if a.state == sm.state and a.isError == sm.isError \
```

Comment:
I'd go ahead and split this across another lines.

if (cond 1 and
    cond2 and
    cond3):
    return True